### PR TITLE
feat: add spring sample to demo ES views

### PR DESCRIPTION
**Pull request by octonato**
https://github.com/lightbend/kalix/issues/7473

### CODE REVIEW

File: samples/spring-eventsourced-customer-registry/src/it/java/customer/api/CustomerIntegrationTest.java [hunk 1]
```diff
@@ -0,0 +1,177 @@
...
+        webClient.post()
+            .uri("/customer/" + id + "/create")
+            .bodyValue(customer)
+            .retrieve()
+            .toEntity(String.class)
+            .block(timeout);
+
+    Assertions.assertEquals(HttpStatus.OK, response.getStatusCode());
+
+    // 4the view is eventually updated
```
Comment (efgpinto) [inline samples/spring-eventsourced-customer-registry/src/it/java/customer/api/CustomerIntegrationTest.java:153]: fat fingers?
```suggestion
    // the view is eventually updated
```

File: samples/spring-eventsourced-customer-registry/src/main/java/customer/api/Address.java [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+
+public class Address {
+  public String street;
+  public String city;
+
+  @JsonCreator
+  public Address(@JsonProperty("street") String street, @JsonProperty("city") String city) {
```
Comment (efgpinto) [inline samples/spring-eventsourced-customer-registry/src/main/java/customer/api/Address.java:14]:
```suggestion
  public Address(String street, String city) {
```

Brave enough to remove these then? :D

File: samples/spring-eventsourced-customer-registry/src/main/java/customer/api/Customer.java [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+
+  // TODO: remove JsonCreator and JsonProperty
+  // this should not be needed and it's not when running the application
+  // however, the integration tests seems to need it. 
+  // Probably related to how the compiler is configured for the tests?
+  @JsonCreator
+  public Customer(@JsonProperty("customerId") String customerId,
+                  @JsonProperty("email") String email,
+                  @JsonProperty("name") String name,
+                  @JsonProperty("address") Address address) {
```
Comment (efgpinto) [inline samples/spring-eventsourced-customer-registry/src/main/java/customer/api/Customer.java:23]:
```suggestion
  public Customer(String customerId,
                  String email,
                  String name,
                  Address address) {
```
Shall we try?

File: samples/spring-eventsourced-customer-registry/src/main/java/customer/api/CustomerEntity.java [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+@RequestMapping("/customer/{id}")
+public class CustomerEntity extends EventSourcedEntity<Customer> {
+
+  private final String entityId;
+
+  public CustomerEntity(EventSourcedEntityContext context) {
+    this.entityId = context.entityId();
+  }
+
+  @GetMapping()
```
Comment (efgpinto) [inline samples/spring-eventsourced-customer-registry/src/main/java/customer/api/CustomerEntity.java:36]:
```suggestion
  @GetMapping
```

File: samples/spring-eventsourced-customer-registry/src/main/java/customer/view/CustomerView.java [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+
+import java.util.Optional;
+
+public class CustomerView {
+  final public String email;
+  final public String name;
+  final public Address address;
+
+  @JsonCreator
+  public CustomerView(@JsonProperty("email") String email, @JsonProperty("name") String name, @JsonProperty("address") Address address) {
```
Comment (efgpinto) [inline samples/spring-eventsourced-customer-registry/src/main/java/customer/view/CustomerView.java:15]:
```suggestion
  public CustomerView(String email, String name, Address address) {
```
going all-in at this point...

**Review by efgpinto — APPROVED** [review]
LGTM, left some minor clean-up suggestions. Hope they don't break CI. 🤞

File: samples/spring-eventsourced-customer-registry/src/it/java/customer/api/CustomerIntegrationTest.java (hunk 1 above)
Comment (octonato) [reply samples/spring-eventsourced-customer-registry/src/it/java/customer/api/CustomerIntegrationTest.java:153]: new keyboard layout :-)

**Comment by octonato** [comment]
@efgpinto, I had to revert the removal of `@JsonProperty`

**Pull request merged**